### ISSUE TITLE
Having trouble while maintaining each dependency as separate module in our project

### ISSUE THREAD

**Issue by zfrc** [mozilla-mobile/android-components#5033]
Hi,

We're using **Firefox-Preview** as our base project and we moved some of android components dependencies as separate modules into our project but we're having trouble while maintaining each module due to replacing module classes with your changes.

We want to move these module into our private repository in order to be able to maintain modules ourself by versioning them and then we want to get commits from your side by adding a remote url to each repo but we couldn't find a remote url for each module.(modules are not compatible to run as separate project)

In a nutshell, is it possible to maintain each component seperately?

**Comment by pocmo** [comment]
Hey!

First of all, this is not an approach we officially support. We recommend that projects like Fenix use our released and versioned artifacts from our maven repository. We try to create APIs that are supporting a variety of use cases and are interested in expanding support for customization.

That said I can understand that there are times were you need to maintain your own fork. Note that whatever solution you pick here: It won't be that smooth and you will have to deal with some pain points that you will have to solve on your own.

I think what you should try in your case is setting up a *composite build* in gradle. Because that will allow you to use both repositories layouts (easy to update) and keep them next to each other.

This PR gives you some hints how to set something like that up:
https://github.com/mozilla-mobile/reference-browser/pull/815/files

In that example I inject an AC checkout into my Reference Browser build and substitute the components in it with my own local version.

With such a setup you should be able to keep the folder structure of your app and the components as the upstream projects and it should be easy to update them normally.

You'd still need two repositories though. However with that setup they will act as one single project in Android Studio.

Does that help?
